File: examples/benchmarks/bench_all.py
import time

# Loop
start = time.time()
for _ in range(50000000):
    pass
print(f"Loop Time: {time.time() - start}")

# Fib
def fib(n):
    if n < 2: return n
    return fib(n-1) + fib(n-2)

start = time.time()
res = fib(25)
print(f"Fib(25) Result: {res}")
print(f"Fib Time: {time.time() - start}")

# Array
data = []
start = time.time()
for i in range(1000000):
    data.append(i)
print(f"Array Len: {len(data)}")
print(f"Array Time: {time.time() - start}")

File: examples/benchmarks/bench_all.js
// Node.js Benchmark
const { performance } = require('perf_hooks');

// Loop
let start = performance.now();
for(let i=0; i<50000000; i++) {}
console.log(`Loop Time: ${(performance.now() - start)/1000}`);

// Fib
function fib(n) {
    if (n < 2) return n;
    return fib(n-1) + fib(n-2);
}
start = performance.now();
let res = fib(25);
console.log(`Fib(25) Result: ${res}`);
console.log(`Fib Time: ${(performance.now() - start)/1000}`);

// Array
let data = [];
start = performance.now();
for(let i=0; i<1000000; i++) {
    data.push(i);
}
console.log(`Array Len: ${data.length}`);
console.log(`Array Time: ${(performance.now() - start)/1000}`);
